packages/ics721/src/execute.rs

```diff
@@ -764,6 +740,40 @@ where
             .add_messages(mint))
     }
 
+    fn mint_msg(
+        &self,
+        token_id: String,
+        token_uri: Option<String>,
+        owner: String,
+        data: Option<Binary>,
+    ) -> StdResult<Binary> {
+        // parse token data and check whether it is of type NftExtension
+        let extension: Option<NftExtensionMsg> = match data {
+            Some(data) => from_json::<NftExtension>(data)
```

Comment:
If we put that other parse in a function I think we could re-use it here?

Reply:
No, we cant, sg-ics721 requires struct of type `Metadata`, whilst ics721 requires `NftExtensionMsg`

Reply:
Out of curiosity how similar are those messages?

Reply:
There are 100% the same. Create a new issue here: https://github.com/public-awesome/launchpad/issues/695

Reply:
Great, thanks